### PR TITLE
Embedding Variable Implement.

### PR DESCRIPTION
**Pull request by candyzone**
# Description

In Large scale learning, It stores embedding data by EmbeddingVariable instead of tf.Variable. We define a set of ops to connect EV ops.

Issue: https://github.com/tensorflow/tensorflow/issues/19324

rfc: https://docs.google.com/document/d/1odez6-69YH-eFcp8rKndDHTNGxZgdFFRJufsW94_gl4/edit?usp=sharing

### CODE REVIEW

File: tensorflow_recommenders_addons/embedding_variable/core/ops/ev_ops.cc [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    .Input("global_step: Tstep")
+    .Attr("T: numbertype")
+    .Attr("Tindices: {int32, int64}")
+    .Attr("Tstep: {int32, int64}")
+    .Attr("use_locking: bool = false")
+    .SetShapeFn(EVApplyGradientDescentShapeFn)
+    .Doc(R"doc(
+)doc");
+
+// TODO
```
Comment (rhdong) [inline tensorflow_recommenders_addons/embedding_variable/core/ops/ev_ops.cc:192]: add @yourname

File: tensorflow_recommenders_addons/embedding_variable/python/__init__.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""EmbeddingVariable."""
+
+import os
+import ctypes
```
Comment (rhdong) [inline tensorflow_recommenders_addons/embedding_variable/python/__init__.py:18]: Case sort ？

**Comment by candyzone** [comment]
@rhdong

File: rfcs/20191101-embedding-variable.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+## Problem & Motivation
+
+"In TensorFlow, we model all data as tensors(n-dimensional arrays) …... At the lowest level, all TensorFlow tensors are dense…... " which quote from TensorFlow white paper<sup>[1]</sup>  states that Tensor is the basic data structure in TensorFlow. Every component(Operation, Rendervous, BuddleReader, BuddleWriter etc.) in TF framework depends on Tensor. It works well with simple interface, clear hierarchy, elegent design.
+
+​    In large scale sparse data training, `embedding_lookup` is the first layer in train graph, the data storage and transmission is rigid. Usually, users invoke the API and build graph as below:
+
+```python
+...
+col0 = tf.contrib.layers.sparse_column_with_hash_bucket(column_name = "col0",
```
Comment (rhdong) [inline rfcs/20191101-embedding-variable.md:18]: `tf.contrib` has been remove from core, maybe some changes needed.

Comment (candyzone) [reply rfcs/20191101-embedding-variable.md:18]: yes, the rfc will be update soon

**Review by liutongxuan — APPROVED** [review]

File: tensorflow_recommenders_addons/embedding_variable/core/kernels/embedding_var.h [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        val = new_val;
+      }
+    }
+    // TODO update ev-version using global_step
+    return val;
+  }
+
+  Status DoLookup(K key, V** val) {
+    tf_shared_lock l(mu_);
+    auto iter = dense_hash_map_.find(key);
```
Comment (dazheyizu) [inline tensorflow_recommenders_addons/embedding_variable/core/kernels/embedding_var.h:91]: 无锁是用cas实现的么

**Pull request merged**